### PR TITLE
impl Error for EncodeUtf8Error, and add a fallback description method

### PR DESCRIPTION
**Pull request by burtonageo**

### CODE REVIEW

File: src/char.rs [hunk 1]
```diff
@@ -25,6 +25,31 @@ const MAX_THREE_B: u32 =  0x10000;
...
+#[cfg(feature = "std")]
+impl ::std::error::Error for EncodeUtf8Error {
+    fn description(&self) -> &str {
+        ERROR_DESCRIPTION
+    }
+}
+
+#[cfg(not(feature = "std"))]
+impl EncodeUtf8Error {
+    fn description(&self) -> &str {
```
Comment (bluss) [inline src/char.rs]: This method seems to be extra, it's not needed?

Comment (burtonageo) [inline src/char.rs]: The [Error](https://doc.rust-lang.org/std/error/trait.Error.html) trait is a std-only trait. I added this method as a fallback for when std is not enabled.

**Comment by bluss** [comment]
Right, it wasn't pub so I didn't understand. The intrinsic method must not be inversely conditional, that is not compatible with crate versioning and features. "std" is an additive feature and can not take anything away. Could just just put the string definition without const inside the intrinsic description method? Thanks

**Comment by bluss** [comment]
Thanks

**Pull request merged**